src/server/controllers/feed.controller.ts

```diff
@@ -85,12 +94,67 @@ export const deleteFeedHandler = async ({
   }
 };
 
+export const syncArticleHandler = async ({
+  id,
+  url,
+}: SyncArticleInput & { id: number }) => {
+  try {
+    const user = await getUserById({ id });
+
+    if (!user) {
+      throw new ApiError(HTTP_STATUS_CODE.NOT_FOUND, "User not found!");
+    }
+
+    if (!user.device) {
+      console.log(`Device not found for user ${user.email ?? user.id}!`);
+      throw new ApiError(
+        HTTP_STATUS_CODE.NOT_FOUND,
+        "Device not found, register it first!",
+      );
+    }
+
+    const api = await getApi({ token: user.device.token });
```

Comment:
I think we should here `Promise.all`.
Like:
Promise.all([getApi({ token: user.device.token ).then((api) => syncArticle({ article: { link: url }, api }))])

Reply:
Hello, I didn't see that promises are related. Thanks @Bartek532 :)